.github/workflows/ci.yml

```
@@ -134,10 +134,18 @@ jobs:
         shell: bash
         run: echo ::set-output name=flavor::$([[ "${{ matrix.os }}" == "cuda" ]] && echo "Cuda" || echo "CPU")
 
+      - name: Set environment variables
+        id: env
+        shell: bash
+        run: |
+          echo "::set-output name=json::{\"ILGPU_CLEAN_TESTS\":\"false\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"1\"}"
+          if [[ ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }} ]]; then
+            echo "::set-output name=json::{\"ILGPU_CLEAN_TESTS\":\"true\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"4\"}"
+          fi
```

ISSUE: These settings do not look correct. The initial conversation says to set clean tests to false, and parallelization level to 4.

We might also want to add some comments about why we are doing this, rather than just what we are doing.

```suggestion
          echo "::set-output name=json::{\"ILGPU_CLEAN_TESTS\":\"false\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"4\"}"
          if [[ ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }} ]]; then
            echo "::set-output name=json::{\"ILGPU_CLEAN_TESTS\":\"true\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"1\"}"
          fi
```

---

agreed about both — and as I mentioned [in a previous review comment](https://github.com/m4rs-mt/ILGPU/pull/712#discussion_r768567426), I think the default value should be an empty json object instead, so that we don't need to change the defaults both in the code and in the CI.

```suggestion
          echo "::set-output name=json::{}"
          if [[ ${{ github.event_name == 'schedule' || (github.event_name == 'push' && (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v'))) }} ]]; then
            echo "::set-output name=json::{\"ILGPU_CLEAN_TESTS\":\"true\", \"ILGPU_TEST_PARALLELIZATION_LEVEL\":\"1\"}"
          fi
```